### ISSUE TITLE
breaking change : multithreading on windows

### ISSUE THREAD

**Issue by Makio64** [yinguobing/head-pose-estimation#2]
An issue appear since the multithreading update on windows 10.

cv2.dnn cant be pickle during the thread creation.

in mark_detector.py :
self.face_net = cv2.dnn.readNetFromCaffe(dnn_proto_text, dnn_model)

Related stackoverflow : https://stackoverflow.com/a/11985056

**Comment by yinguobing** [comment]
Yes, `multithreading` is imported to speed up the entire process. However it seems not working on macOS High Sierra which I've tried, and the issue seems to be with OpenCV itself, maybe TBB related.

As there is no Windows PC at hand, I'm afraid Windows users need to figure it out by themselves. Or, this [commit](https://github.com/yinguobing/head-pose-estimation/tree/b5136aa79f14016c2eb604508542a3c0f0f4bfd5) without multithreading could be helpful.

**Comment by Makio64** [comment]
Yes, I posted the issue to help futur users to find the commit you indicated below more than getting a support/solution.

Maybe an OS checking or flag to activate or not multithreading will be a easy solution to put on.

PS: Thanks for your work, it helped me a lot! 👍

**Comment by yinguobing** [comment]
OS checking is a great idea which I should have thought of!

This feature will be added in the next commit. Also please feel free to send any pull request.

**Comment by ins2718** [comment]
Hi everyone. In advance, I'm sorry for bad English. What about creating a thread instead of a process? Something like this (It works on windows):
```python
import threading
#...
class get_face(object):
    def __init__(self, detector, img_queue, box_queue):
        thread = threading.Thread(target=self.run, args=(detector, img_queue, box_queue))
        thread.daemon = True
        thread.start()
    def run(self, detector, img_queue, box_queue):
        while True:
            image = img_queue.get()
            box = detector.extract_cnn_facebox(image)
            box_queue.put(box)
#...
    img_queue.put(sample_frame)
    get_face(mark_detector, img_queue, box_queue)
```

**Comment by yinguobing** [comment]
@ins2718 Glad you make it work! I'm sure this is helpful for other Windows users.

**Comment by humphreyBristol** [comment]
@flyyufenfei123 @yinguobing  Since as ins2718 said we can replace process with thread. But in the code, there still exists Queue from multiprocessing, and how should we deal with it?

**Comment by yinguobing** [comment]
I guess a pull request will make everybody happy😃

**Comment by elhammoumi** [comment]
@ins2718 your Thread solution is working, I also got an error in `mark_detector.py` in 
`self.face_net.setInput(cv2.dnn.blobFromImage(
            image, 1.0, (300, 300), (104.0, 177.0, 123.0), False))`
I removed the last parameter. I don't know what it does, I just read online that this method takes max 5 params.
and I commented in `estimate_head_pose.py` the line `print(tm.getTimeSec()/tm.count())`
